src/GraphQL/Types/Scalars/BigIntGraphType.cs

```diff
@@ -10,22 +10,20 @@ namespace GraphQL.Types
     public class BigIntGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object? ParseLiteral(IValue value) => value switch
+        public override object? ParseLiteral(GraphQLValue value) => value switch
         {
-            IntValue intValue => new BigInteger(intValue.Value),
-            LongValue longValue => new BigInteger(longValue.Value),
-            BigIntValue bigIntValue => bigIntValue.Value,
-            NullValue _ => null,
+            GraphQLFloatValue x => BigInt.Parse(x.Value),
+            GraphQLIntValue x => BigInt.Parse(x.Value),
+            GraphQLNullValue _ => null,
             _ => ThrowLiteralConversionError(value)
         };
 
         /// <inheritdoc/>
-        public override bool CanParseLiteral(IValue value) => value switch
+        public override bool CanParseLiteral(GraphQLValue value) => value switch
         {
-            IntValue _ => true,
-            LongValue _ => true,
-            BigIntValue _ => true,
-            NullValue _ => true,
+            GraphQLFloatValue x => BigInt.TryParse(x.Value, out var _),
```

Comment:
BigInteger need not parse floats.  Literals without a period will be an int value, and literals with a period will never parse.

Reply:
I thought about it, I'll remove float case for all integer graph types.

Reply:
Removed from all integer graph types.